### ISSUE TITLE
Time performance issue

### ISSUE THREAD

**Issue by MaToxZ** [sarxos/webcam-capture#629]
Hi guys,

Congratulations for this library, it so helpful, quick and easy to use.

This time I came here to ask you if it is a normal behavior that takes to long (about 5-8sec) to open a webcam view. Right now I'm developing and a software for visitors registration, but each time a user select the "Entrance Registration" option, It does takes too long to show the webcam view, maybe am I doing something wrong? I'll Appriciate if you guys can guide me, or if it is a normal behavior about this library.

By the way, I want to ask you, how do I finish the Webcam view at all, I mean, I already used the webcam, took my snapshot, now how I close it without closing the entire frame. thank you.

**Comment by sarxos** [comment]
Hi @MaToxZ,

What is the operating system this application is running on?

In regards to:

> takes to long (about 5-8sec) to open a webcam view
> how do I finish the webcam view at all

Can you please define what _webcam view_ is in your case? Are you referring to `WebcamPanel` or something else?

If you are not doing any `Thread.sleep()` in your code or some blocking operations to e.g. query database, you should get your camera online in ~1-2 seconds average. This, however, depends on the operating system, computer hardware and the camera itself. There is no golder rule here, but you can test this out by:

a) Using different camera,
b) Running your application on a different (faster) computers,

If neither a) and b) help and your webcam view is still opening so long, then you should consider that you may have a bug in your code which causes this delay.

**Comment by MaToxZ** [comment]
Yes @sarxos I'm sure you're right, the point is I am kind of Beginner, I'm not pretty sure about how implement the "Thread", I was thinking to launch the WebCam View (WebcamPanel) in a thread but I don't know how to do this.

I have JFrame form where I fill the fields, one of these fields should be a picture, that's why I created a JPanel and I JPanel.add(WebcamPanel). When I try to open this JFrame, It does now take like 5-8sec, It didn't happen before. so that's why I suggest It is the Webcam which is taking that long. could you help me how to achieve this? (Open the Jframe, add the WebcamPanel to my JFrame>JPanel, as fast as possible)?

**Comment by pequierda** [comment]
sir i have concern  when i close the camera the java is running in my computer in the task manager how to fix it here's my code 

JButton btnCloseCamera = new JButton("close camera");
		btnCloseCamera.addActionListener(new ActionListener() {
			public void actionPerformed(ActionEvent arg0) {
				
			
				new Thread()
				{
					
					
				
					public void run()
					{
						
						webcam.close();
						img_icon.setIcon(null);
						
						webcam.resetDriver();
					}
				}.start();
				
			}
		});
		btnCloseCamera.setBounds(730, 198, 96, 29);
		contentPane.add(btnCloseCamera);

when i push that button i have this stacktrace 
13660 [Thread-5] WARN com.github.sarxos.webcam.Webcam - Cannot get image, webcam has been already disposed

and it infinite running until i terminate it in my IDE

**Comment by pequierda** [comment]
i have this prob because in my backgroud it keeps running :(
